### ISSUE TITLE
We need some convenient space types

### ISSUE THREAD

**Issue by zsunberg** [JuliaPOMDP/POMDPs.jl#301]
Right now it is inconvenient to implement continuous action spaces. We need some convenient space types, like `Box` or something. Unclear if this already exists somewhere or it should go into `POMDPModelTools` or some new package named with some combination of the words "Box" "Interval" "Real" "Multidimensional" "Vector"

**Comment by MaximeBouton** [comment]
Are you specifically thinking about continuous spaces? 
This might be useful: https://github.com/invenia/Intervals.jl

**Comment by lassepe** [comment]
Are simple sets like `Box` enough? Or do we need support for arbitrary polytopes? It sounds like sampling may become rather tedious at some point.

**Comment by zsunberg** [comment]
> This might be useful: https://github.com/invenia/Intervals.jl

Yes, something like that that someone has already implemented would be great! There are two problems with that particular package though:
1) It doesn't have `rand` implemented
2) It doesn't work like we would want:
<pre><font color="#8AE234"><b>julia&gt; </b></font>[1.5,2.5] in Interval([1,1], [2,2])
true
</pre>

Are there any other packages?

> Are simple sets like Box enough?

Yes, I think boxes are enough for now! if people want polytopes they can write their own space objects easily enough. 

The other thing that we might want is cartesian products, i.e. [0,1] x {true, false}

**Comment by lassepe** [comment]
Maybe we should just make a small implementation for ourselves since hybrid spaces might be already quite a special needs. Product distributions are already supported by `Distribitions.jl`so the `rand` interface should be easy to implement for these simple spaces.

**Comment by MaximeBouton** [comment]
There is this as well that we could take inspiration from: https://github.com/FluxML/Gym.jl/tree/master/src/Spaces 

we could also use that directly but I don't know if we want a dependency on Gym.jl
